test cases/common/193 dict/meson.build

```diff
@@ -33,3 +33,35 @@ d3 = d2
 d3 += {'e' : 'f'}
 assert(d3 == {'a' : 'b2', 'c' : 'd', 'e' : 'f'}, 'dict plusassign is not working')
 assert(d2 == {'a' : 'b2', 'c' : 'd'}, 'dict should be immutable')
+
+dict1 = {}
+
+# A variable to be used as a key
+testkey1 = 'myKey1'
+testkey2 = 'myKey2'
+
+# Add new entry using the variable
+dict1 += {testkey1 : 'myValue'}
+dict1 += {testkey2 : 42}
+
+# Test that the stored values are correct
+assert(dict1[testkey1] == 'myValue',
+       'Incorrect string value retrieved from dictionary - variable key')
+assert(dict1['myKey1'] == 'myValue',
+       'Incorrect string value retrieved from dictionary - literal key')
+assert(dict1[testkey2] == 42,
+       'Incorrect int value retrieved from dictionary - variable key')
+assert(dict1['myKey2'] == 42,
+       'Incorrect int value retrieved from dictionary - literal key')
+
+d = {testkey1 : 1}
+assert(d[testkey1] == 1,
+       'Incorrect int value retrieved from dictionary - variable key')
+assert(d['myKey1'] == 1,
+       'Incorrect int value retrieved from dictionary - literal key')
+
+d = {'1' / '2' : 1, join_paths('a', 'b') : 2}
```

Comment:
just for completeness could you add:
```meson
d['a' + 'b'] = 3
assert(d['a' + 'b'] == 3, 'Incorrect expression evaluation in dictionary key')
```

Reply:
We don't support `d[key] = value` syntax because dictionaries are immutable. The equivalent is `d += {'a' + 'b':3}` which does not modify d, it replace it.

But still added unit test for + operator in dict key.